File: 4_Klassen/Objective-C/Bucketlist/Bucketlist/IOHelper.m
//
//  IOHelper.m
//  Bucketlist
//
//  Created by Jan Brinkmann on 28/06/16.
//  Copyright © 2016 Jan Brinkmann. All rights reserved.
//

#import "IOHelper.h"
#import <readline/readline.h>

@implementation IOHelper

- (NSString *)newInput
{
    printf("Willkommen! Gib 'h' für Infos zur Nutzung ein\n");
    
    NSString *input = [self ask:@":> "];
    return input;
}

- (NSString *)ask:(NSString *)theQuestion
{
    printf("%s", [theQuestion UTF8String]);
    return [self getInput];
}

- (NSString *)usageInfo
{
    return [NSString stringWithFormat:@"%@\n%@\n%@\n%@\n%@\n", @"Bitte wählen:",
        @"h: diese Hilfe anzeigen",
        @"a: hinzufügen",
        @"r: entfernen",
        @"l: list\n"];
}

- (NSString *)getInput
{
    NSString *input = [NSString stringWithUTF8String:readline(NULL)];
    
    return input;
}
@end

File: 4_Klassen/Swift/Bucketlist/Bucketlist/IOHelper.swift
//
//  IOHelper.swift
//  Bucketlist
//
//  Created by Jan Brinkmann on 28/06/16.
//  Copyright © 2016 Jan Brinkmann. All rights reserved.
//

import Foundation

class IOHelper {
    func newInput() -> String? {
        print("Willkommen! Gib 'h' für Infos zur Nutzung ein")
        
        let input = ask(":> ")
        return input
    }
    
    func ask(theQuestion: String) -> String? {
        print("\(theQuestion)", terminator: "")
        
        return getInput()
    }
    
    func getInput() -> String? {
        let input = readLine(stripNewline: true)
        return input
    }
    
    func usageInfo() -> String {
        return
            "Bitte wählen:\n"
                + "h: diese Hilfe anzeigen\n"
                + "a: hinzufügen\n"
                + "r: entfernen\n"
                + "l: list\n"
    }
}